daxgen/daxgen.py

```diff
@@ -164,6 +161,20 @@ def _check_sanity(self):
             except nx.NetworkXError:
                 raise ValueError("Graph is not bipartite.")
 
+            U, V = nx.bipartite.sets(self.graph)
+
+            # Select an arbitrary vertex from the set U and based on its
+            # attributes decide if the set U represents files and respectively,
+            # V represents tasks, or the other way round.
+            v = next(iter(U))
+            node_attrs = set(self.graph.node[v].keys())
+            files, tasks = (U, V) if node_attrs >= _FILE_ATTRS else (V, U)
```

Comment:
is this the only use of _FILE_ATTRS? I mean you used that attribute before 'lfn' in other parts of the code.

Why not just `if _FILE_ATTRS.issubset(node_attrs)` ?